.github/workflows/build-ros2-ws-action.yml

```
@@ -0,0 +1,79 @@
+name: CI
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches: [master]  
+jobs:
+  build-and-test:
+    name: Build and Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4.1.1
+        with:
+          path: aruku
+          ref: feature/fix-using-grpc
+
+      - name: Checkout Kansei
+        uses: actions/checkout@v4.1.1
+        with:
+          repository: ichiro-its/kansei          
+          path: kansei
+
+      - name: Checkout Kansei Interfaces
+        uses: actions/checkout@v4.1.1
+        with:
+          repository: ichiro-its/kansei_interfaces
+          path: kansei_interfaces
+      
+      - name: Checkout Aruku Interfaces
+        uses: actions/checkout@v4.1.1
+        with:
+          repository: ichiro-its/aruku_interfaces
+          path: aruku_interfaces
+
+      - name: Checkout Keisan
+        uses: actions/checkout@v4.1.1
+        with:
+          repository: ichiro-its/keisan
+          path: keisan
+
+      - name: Checkout tachimawari
+        uses: actions/checkout@v4.1.1
+        with:
+          repository: ichiro-its/tachimawari
+          path: tachimawari
+
+      - name: Checkout tachimawari interfaces
+        uses: actions/checkout@v4.1.1
+        with:
+          repository: ichiro-its/tachimawari_interfaces
+          path: tachimawari_interfaces
+
+      - name: Setup cmake        
+        run: sudo apt install -y build-essential autoconf libtool pkg-config cmake
+
+      - name: Setup abseil for grpc              
+        run: |
+          git clone https://github.com/abseil/abseil-cpp.git
+          mkdir -p abseil-cpp/build && cd  abseil-cpp/build
+          cmake ..
+          make -j 4
+          sudo make install
+      
+      - name: Setup grpc        
+        run: |
+          git clone --recurse-submodules -b v1.61.0 --depth 1 --shallow-submodules https://github.com/grpc/grpc
+          cmake -B abseil-cpp/build
+          cmake --build abseil-cpp/build
+          cmake --install abseil-cpp/build
```

It seems that you put this wrongly, instead of in the `Setup grpc` step, it should be in the `Setup abseil for grpc` step. But anyway there's also a bug in here, use this instead:
```suggestion
          cmake -S grpc -B grpc/build
          cmake --build grpc/build
          cmake --install grpc/build
```